### PR TITLE
update signal iterator

### PR DESCRIPTION
**Pull request by tluebeck**
### Which issue(s) are closed by this pull request?

Closes #

### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

File: tests/test_audio_signal_iterator.py [hunk 1]
```diff
@@ -17,6 +17,20 @@ def test_simple_iter():
         npt.assert_array_equal(s._data, np.atleast_2d(data[idx, :]))
 
 
+def test_complex_simple_iter():
```
Comment (f-brinkmann) [inline tests/test_audio_signal_iterator.py:20]:
```suggestion
@pytest.mark.parametrize('is_complex', [False, True])
def test_complex_simple_iter():
```

File: tests/test_audio_signal_iterator.py [hunk 2]
```diff
@@ -17,6 +17,20 @@ def test_simple_iter():
...
+    data = np.ones((3, 1024)) * np.atleast_2d(np.arange(3)).T
+    sig = Signal(data, 1)
+    sig.complex = True
```
Comment (f-brinkmann) [inline tests/test_audio_signal_iterator.py:23]:
```suggestion
    sig.complex = is_complex
```

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks for catching. I would suggest to parametrize the existing test to minimize duplicate code

File: tests/test_audio_signal_iterator.py (hunk 1 above)
Comment (tluebeck) [reply tests/test_audio_signal_iterator.py]: good point thank you :)

**Pull request merged**